### PR TITLE
Mattray gov/appeals 10276 lint

### PR DESCRIPTION
**Pull request by mikefinneran**
- https://vajira.max.gov/browse/APPEALS-10276
- https://vajira.max.gov/browse/APPEALS-10277

Creates [find_document_version_reference_by_date_range.rb](https://github.com/department-of-veterans-affairs/connect_vbms/commit/2d77d752538b11e93eabec504512b88b49ec75a4#diff-561229b3a047754f5a927e17e8c60649da2a1be9ba27cd4f3dea79948fbef07f) request to query for documents that have a vbmsUploadDate between the range provided to the method.

### CODE REVIEW

File: spec/requests/find_document_version_reference_by_date_range_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      xml = fixture("responses/find_document_version_reference_by_date_range_response.xml")
+      doc = parse_strict(xml)
+      @vbms_docs = request.handle_response(doc)
+    end
+    subject { @vbms_docs }
+    it "should return an array of documents with an upload date within the date range" do
+      expect(subject.map { |doc| Time.new(2017, 0o3, 22).to_datetime <= doc.upload_date.to_datetime }).to all(be_truthy)
+      expect(subject.map { |doc| Time.new(2017, 0o3, 29).to_datetime >= doc.upload_date.to_datetime }).to all(be_truthy)
+      expect(subject).to be_an(Array)
+      expect(subject.count).to be 7
```
Comment (mikefinneran) [inline spec/requests/find_document_version_reference_by_date_range_spec.rb:32]: Avoid "magic numbers and "magic strings" by defining constants in a constant file if applicable or at the top of the test.
- https://eliotsykes.com/2015/08/18/magic-numbers/

The pattern in the rest of the request file tests is using magic numbers and strings as well as seen below so we will continue but refactoring many of these tests could be useful in the future.

- https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/spec/requests/find_document_series_reference_spec.rb
- https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/spec/requests/list_contentions_spec.rb
- https://github.com/department-of-veterans-affairs/connect_vbms/blob/master/spec/requests/ for the remainder

File: lib/soap_scum/ws_security.rb [hunk 2]
```diff
@@ -105,9 +105,9 @@ def decrypt(xml)
       def generate_block_cipher
         case @cipher_algorithm
         when CryptoAlgorithms::AES128
-          OpenSSL::Cipher::AES128.new(:CBC)
+          OpenSSL::Cipher.new("aes-128-cbc")
         when CryptoAlgorithms::AES256
-          OpenSSL::Cipher::AES256.new(:CBC)
```
Comment (jshields13) [inline lib/soap_scum/ws_security.rb:110]: What does this change do?

**Review by jshields13 — CHANGES_REQUESTED** [review]
PR description needs to have the linked jira item and Test plan in jira also side note comments are a programmers friend

File: lib/soap_scum/ws_security.rb (hunk 2 above)
Comment (mattray-gov) [reply lib/soap_scum/ws_security.rb:110]: This was a linter fix.

**Review by mattray-gov — APPROVED** [review]
LGTM

**Pull request merged**